tests/integration/components/polaris-connected-test.js

```diff
@@ -0,0 +1,83 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import buildNestedSelector from '../../helpers/build-nested-selector';
+
+const leftContent = 'left content';
+const rightContent = 'right content';
+const yieldedContent = 'yielded content';
+const connectedContainerSelector = '.Polaris-Connected';
+const primaryContentSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--primary');
+
+// These selectors are for when both `left` AND `right` attributes are passed-in
+const leftContentSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--connection:first-of-type');
+const rightContentSelector = buildNestedSelector(connectedContainerSelector, '.Polaris-Connected__Item--connection:last-of-type');
+
+module('Integration | Component | polaris-connected', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it does not render any connected elements if `left` or `right` attributes are missing', async function(assert) {
+    this.setProperties({
+      primary: yieldedContent
+    });
+
+    await render(hbs`
+      {{#polaris-connected}}
+        <div class="primary-content">
+          {{primary}}
+        </div>
+      {{/polaris-connected}}
+    `);
+
+    const primaryContentSelector = '.primary-content';
+
+    assert.dom(connectedContainerSelector).doesNotExist('connected component container does not get rendered');
+    assert.dom(leftContentSelector).doesNotExist('left content node does not get rendered');
+    assert.dom(rightContentSelector).doesNotExist('right content node does not get rendered');
+    assert.dom(primaryContentSelector).exists('the block content still renders when connected sides are not passed in');
```

Comment:
nit: this `exists` test is basically also tested by the `hasText` below (same below)

Reply:
Hm I'd agree if the `primaryContentSelector` wasn't nested inside a conditional in the component template. In this case there's both a chance that the element may or may not get rendered, as well as the chance that the value inside the element could be wrong.

Only testing `hasText` and having it fail might lead you to believe the element itself never got rendered, when it could just be that the value got wrongly mutated somehow. Although now I'm also just being nitpicky because I'm sure the error message would be pretty explicit 😝

Reply:
yeah, no worries...fine with this.....but yeah, that's what I'm ...this will catch your error whatever that is, cause you can't check a text if the element doesn't exist